### ISSUE TITLE
Asustor: https fails with latest Asustor Mono package

### ISSUE THREAD

**Issue by NomisTuo** [Jackett/Jackett#3489]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue. Duplicating issues makes it more difficult for everyone to follow. Your issue may have already been solved in the past as well.

Provide a description of the feature request or bug, the more details the better.
If you are experiencing an issue with a tracker, a **full enhanced log must be included**. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting

**Jackett version**:  0.9.16.0
**Mono version** (if not using Windows): 5.8.1.0.r9 (Asustor)  

-------------------------------
The gktorrent, isohunt2, cpasbien and nextTorrent indexers are sending the Following error (Exception (gktorrent): The authentication or decryption has failed.: An error occurred while sending the request).

Here is the beginning  error entry in the log file.
Jackett.Common.IndexerException: Exception (gktorrent): The authentication or decryption has failed. ---> System.Net.Http.HttpRequestException: An error occurred while sending the request ---> System.Net.WebException: Error: SecureChannelFailure (The authentication or decryption has failed.) ---> System.IO.IOException: The authentication or decryption has failed. ---> System.IO.IOException: The authentication or decryption has failed. ---> Mono.Security.Protocol.Tls.TlsException: The authentication or decryption has failed.
  at Mono.Security.Protocol.Tls.RecordProtocol.EndReceiveRecord (System.IAsyncResult asyncResult) [0x00049] in /root/asustor/trunk/trunk_g3/source/mono-5.8.1.0/mcs/class/Mono.Security/Mono.Security.Protocol.Tls/RecordProtocol.cs:434 
  at Mono.Security.Protocol.Tls.SslClientStream.SafeEndReceiveRecord (System.IAsyncResult ar, System.Boolean ignoreEmpty) [0x00000] in /root/asustor/trunk/trunk_g3/source/mono-5.8.1.0/mcs/class/Mono.Security/Mono.Security.Protocol.Tls/SslClientStream.cs:256 
  at Mono.Security.Protocol.Tls.SslClientStream.NegotiateAsyncWorker (System.IAsyncResult result) [0x00071] in /root/asustor/trunk/trunk_g3/source/mono-5.8.1.0/mcs/class/Mono.Security/Mono.Security.Protocol.Tls/SslClientStream.cs:418 
   --- End of inner exception stack trace ---
  at Mono.Security.Protocol.Tls.SslClientStream.EndNegotiateHandshake (System.IAsyncResult result) [0x00033] 

Thank you in advance for your help.

Best regards.

NomisTuo

**Comment by kaso17** [comment]
is this still a problem?
What kind of CPU is the system using?

**Comment by NomisTuo** [comment]
Hi kaso17,

Thank you for answering.
The problem is still present for the mentionned indexers. Jackett is working perfectly for others as thepiratebay.

My cpu is: Intel Celeron J3455 Quad-Core 1.5 GHz (brust up 2.3GHz) Processor

Thanks in advance.

**Comment by kaso17** [comment]
try upgrading to mono 5.12

**Comment by NomisTuo** [comment]
Unfortunately the 5.12 is not available on asustor AppStore. Do you know if installing the 5.12 manually could lead to some issue with the asustor adm interface?

Thank you in advance.

**Comment by flightlevel** [comment]
I don't think cert-sync runs automatically with the Asustor mono package
Follow the steps in this issue https://github.com/Jackett/Jackett/issues/3373

**Comment by NomisTuo** [comment]
Hi all,

@kaso, thank you already for your help.
@flightlevel, I already updated the certificates using the cert-sync tool

It looks like it indexer dependent. Is there a way to update the indexer definition of Jackett?

Thank you in advance

**Comment by flightlevel** [comment]
What happens if you run `csharp -e 'Console.WriteLine (new System.Net.WebClient ().DownloadString ("https://www.howsmyssl.com/").IndexOf ("1.2"))'`

Do any https trackers work from Jackett?

**Comment by itdaboyz** [comment]
I have the same error and in the same situation (Asustor NAS and latest Mono version available for us 5.8):

- Xthor and HD-Only are not working.
- Jackett updates are not working (Error checking for updates.).
- Abnormal, Elite Tracker and YGGTorrent are HTTPS trackers and are working.

Could it be something with self signed certificates?

If I run the previous comand, it says nothing but
`>`

**Comment by kaso17** [comment]
`csharp -e 'Console.WriteLine (new System.Net.WebClient ().DownloadString ("https://www.howsmyssl.com/").IndexOf ("1.2"))'` should return a number. Please double check the output.

**Comment by itdaboyz** [comment]
My bad, I did the command without the last '.
It seems that "csharp" is not installed on Asustor.
`-sh: csharp: not found  `

**Comment by Andrufus** [comment]
I have the same problem and the same config(asustor nas) as you guys and for me Idope, TorrentDownloads and Zooqle (all https) are working and 1337x for example is not working with the same error message mentioned in the title of this issue.
I too ran the cert-sync command. And I think it worked fine right after Asus updated Mono. The change has to be in the recent Jackett updates right ?
(I have the 0.9.23 btw)

**Comment by itdaboyz** [comment]
`ps faux | grep Jackett`
```
18801 admin     20:59 mono --debug /usr/local/AppCentral/jackett/Jackett/JackettConsole.exe -d /usr/local/AppCentral/jackett/config                                                           
25726 root       0:00 grep Jackett         
```                                                                                                                                                   

For me, the error happened right after the last Mono update delivered by Asustor.
5.4.1.6.r6 worked great until Jackett needed a more up to date version.
5.8.1.0.r6 worked after a cert-sync and some file modification
5.8.1.0.r9 doens't work.

**Comment by kaso17** [comment]
I'd recommend to report this problem to the asustor community.
This error is caused by the mono HTTP client, seems like there's a problem with the 5.8.1.0.r9.

**Comment by flightlevel** [comment]
Yep, like @kaso17 says its definitely an Asustor issue
But, pretty sure Asustor can run Docker. Getting maintained packages is difficult enough on popular platforms like QNAP and Synology. Its going to be way harder on Asustor. It takes a bit of time to get your head around Docker, but once you do, you'll always have access to the latest packages, no more SSL issues and more packages to choose from

**Comment by itdaboyz** [comment]
Not all Asustor's NAS can run Docker. Old model like mine are blocked on Linux kernel 3.4.26 and as far as I know, Docker needs 3.8 to work.
So we kinda screwed on this and totally dependent of Asustor.

**Comment by ShaMana999** [comment]
Would you be able to provide some reading material on docker and it's use for the specific situation. I started doing it on my own and wen't nowhere. I have got practically no knowledge in Linux operations, so is that method practical for a entry level user? Asustor doesn't seem to be in a hurry to do practically anything so everyone for himself it seems

**Comment by skuizy** [comment]
Same here for me on my odroid XU4 running ubuntu 18.04.1, with mono 5.12.0.301 from official repo.

Doesn't work :
1337x
cpasbien
Downloadville
GkTorrent
RuTracker
Torlock
Torrent9
Torrentz2
World Wide Torrents
Yggtorrent

Work :
Elite-Tracker
EZTV
Idope
LimeTorrents
MagnetDL
The Pirate Bay
Torrent Downloads
Zooqle

I have already synced my certs with cert-sync and when I run the command `csharp -e 'Console.WriteLine (new System.Net.WebClient ().DownloadString ("https://www.howsmyssl.com/").IndexOf ("1.2"))'` the result is 3636 (I have no idea of the meaning of this value btw...)

If I run the test as shown on the mono project help page : `mono tlstest.exe https://www.nuget.org`, I get https://www.nuget.org for response. But when I run the same test with https://www.gktorrent.org/, for exemple, I get this error : 
```
FAILED: #-2146232800
System.IO.IOException: The authentication or decryption has failed. ---> Mono.Security.Protocol.Tls.TlsException: The authentication or decryption has failed.
  at Mono.Security.Protocol.Tls.RecordProtocol.EndReceiveRecord (System.IAsyncResult asyncResult) [0x00037] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at Mono.Security.Protocol.Tls.RecordProtocol.ReceiveRecord (System.IO.Stream record) [0x0000a] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at Mono.Security.Protocol.Tls.SslClientStream.SafeReceiveRecord (System.IO.Stream s) [0x00000] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at Mono.Security.Protocol.Tls.SslClientStream.OnNegotiateHandshakeCallback (System.IAsyncResult asyncResult) [0x0000e] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at Mono.Security.Protocol.Tls.SslStreamBase.AsyncHandshakeCallback (System.IAsyncResult asyncResult) [0x0000c] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
   --- End of inner exception stack trace ---
  at Mono.Security.Protocol.Tls.SslStreamBase.EndNegotiateHandshake (Mono.Security.Protocol.Tls.SslStreamBase+InternalAsyncResult asyncResult) [0x00022] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at Mono.Security.Protocol.Tls.SslStreamBase.NegotiateHandshake () [0x0002b] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at Mono.Security.Protocol.Tls.SslStreamBase.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00064] in <fd664b637bcb4c1990ddfa2289ae7c6a>:0
  at System.IO.StreamWriter.Flush (System.Boolean flushStream, System.Boolean flushEncoder) [0x0007e] in <b0e1ad7573a24fd5a9f2af9595e677e7>:0
  at System.IO.StreamWriter.Flush () [0x00006] in <b0e1ad7573a24fd5a9f2af9595e677e7>:0
  at TlsTest.GetStreamPage (System.String url) [0x000c7] in <caa05ae54de94cf18e288ed2f60de313>:0
  at TlsTest.Main (System.String[] args) [0x00392] in <caa05ae54de94cf18e288ed2f60de313>:0`
```

And if I curl -vvv https://www.gktorrent.org/ I get :
```
*   Trying 104.27.128.5...
* TCP_NODELAY set
* Connected to www.gktorrent.org (104.27.128.5) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-ECDSA-CHACHA20-POLY1305
* ALPN, server accepted to use h2
* Server certificate:
*  subject: OU=Domain Control Validated; OU=PositiveSSL Multi-Domain; CN=sni228562.cloudflaressl.com
*  start date: Aug  3 00:00:00 2018 GMT
*  expire date: Feb  9 23:59:59 2019 GMT
*  subjectAltName: host "www.gktorrent.org" matched cert's "*.gktorrent.org"
*  issuer: C=GB; ST=Greater Manchester; L=Salford; O=COMODO CA Limited; CN=COMODO ECC Domain Validation Secure Server CA 2
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x509138)
> GET / HTTP/2
> Host: www.gktorrent.org
> User-Agent: curl/7.58.0
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 200
...
```

I didn't found any recent issue related to this on mono's github. The only old issues I found all talled to resync certs....

**Comment by skuizy** [comment]
The issue is still present with the 5.14.0.177 update from today.

**Comment by flightlevel** [comment]
@ShaMana999 
It will take you a bit of time
Install this package first
https://www.asustor.com/en/app_central/app_detail?id=754&type=&model=

then read
https://www.smarthomebeginner.com/install-jackett-using-docker/

more docker detail
https://www.smarthomebeginner.com/docker-home-media-server-2018-basic/


@skuizy Your issue is quite different, everyone else on this issue has problems with Asustor's Mono package whereas you are having problems droid/ARM platform
Can you please post the output of running `cert-sync /etc/ssl/certs/ca-certificates.crt` as root/sudo

**Comment by skuizy** [comment]
OK sorry, you might want me to open another issue then ?

```
sudo cert-sync /etc/ssl/certs/ca-certificates.crt
Mono Certificate Store Sync - version 5.14.0.177
Populate Mono certificate store from a concatenated list of certificates.
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Importing into legacy system store:
I already trust 133, your new list has 133
Import process completed.

Importing into BTLS system store:
I already trust 133, your new list has 133
Import process completed.
```

I assume my certs are already up-to-date...

**Comment by flightlevel** [comment]
@skuizy Can't see the point of opening a new issue on Jackett, as `mono tlstest.exe` is failing. This indicates its a Mono issue, potentially ARM64 related??? 
One option would be to open an issue on Mono's repo or potentially try running inside Docker?
Actually, found this, worth a read, but not promising https://github.com/Jackett/Jackett/issues/2575

**Comment by skuizy** [comment]
Thanks for the confirmation, I've opened mono/mono#10050

**Comment by skuizy** [comment]
@flightlevel As stated in the mono thread, my issue was caused by `MONO_TLS_PROVIDER=legacy` in my systemd. 

Changing legacy by btls solved it.

**Comment by ShaMana999** [comment]
I just wanted to share my temporary solution, after updating to the last jacket version and not being able to run Mono through docker functionally. I manually updated Jackett to an earlier version and disabled the auto-update and it seems to work with the indexers that I use. For now seems that my options are to continue my education and figure out how to make the whole thing work manually or wait for asustor to update mono
